File: contract-service/src/main/java/com/example/contract_service/service/ContractService.java
package com.example.contract_service.service;

import com.example.contract_service.dto.CreateContractRequest;
import com.example.contract_service.entity.Contract;
import java.time.LocalDate;
import java.util.List;

public interface ContractService {
    Contract createFull(CreateContractRequest req);
    List<Contract> getAll();
    List<Contract> getByRoom(Long roomId);
    List<Contract> getByTenant(Long tenantId);
    Contract cancel(Long id);
    List<Contract> getActiveContracts(LocalDate date);
    Contract updateFull(Long id, CreateContractRequest req);
    void deleteContract(Long id);
}

File: phongtro-fe/src/api/contractService.js
import api from './axios';

export const contractService = {
  getAll: () => api.get('/contracts'),
  getByRoom: (roomId) => api.get(`/contracts/room/${roomId}`),
  getByTenant: (tenantId) => api.get(`/contracts/tenant/${tenantId}`),
  create: (data) => api.post('/contracts', data),
  cancel: (id) => api.put(`/contracts/${id}/cancel`),
  getActive: (date) => api.get('/contracts/active', { params: { date } }),
};